config/proxy_config.go

```diff
@@ -0,0 +1,120 @@
+package config
+
+import (
+	"fmt"
+
+	"github.com/Layr-Labs/eigenda-proxy/common"
+	"github.com/Layr-Labs/eigenda-proxy/config/eigendaflags"
+	eigendaflags_v2 "github.com/Layr-Labs/eigenda-proxy/config/eigendaflags/v2"
+	"github.com/Layr-Labs/eigenda-proxy/server"
+	"github.com/Layr-Labs/eigenda-proxy/store"
+	"github.com/Layr-Labs/eigenda-proxy/store/generated_key/memstore"
+	"github.com/Layr-Labs/eigenda-proxy/store/generated_key/memstore/memconfig"
+	"github.com/Layr-Labs/eigenda-proxy/verify/v1"
+	"github.com/urfave/cli/v2"
+
+	"github.com/Layr-Labs/eigenda/api/clients"
+)
+
+// ProxyConfig ... Higher order config which bundles all configs for orchestrating
+// the proxy server with necessary client context
+type ProxyConfig struct {
+	ServerConfig        server.Config
+	EdaClientConfigV1   clients.EigenDAClientConfig
+	EdaVerifierConfigV1 verify.Config
+
+	EdaClientConfigV2 common.ClientConfigV2
+
+	MemstoreConfig *memconfig.SafeConfig
+	StorageConfig  store.Config
+
+	// Enabling will turn on memstore for both V1 && V2
+	//
+	// Currently the memstore is not persisted to disk, and is lost when proxy is stopped. Testing migrations is
+	// currently not supported because it requires restarting the proxy with new flags, which will cause the
+	// memstore db to be wiped.
+	//
+	MemstoreEnabled bool
+
+	EigenDAV2Enabled bool
```

Comment:
add documentation for semantic of this flag. What does it do? Does it make puts be sent to v2? Or does it enable v2 reading only?

Reply:
As part of [this issue](https://linear.app/eigenlabs/issue/DAINT-324/implement-v1-v2-canary-testing), I want to entirely rework the interaction between v1 and v2, which will include this flag.